### PR TITLE
Add missions support

### PR DESCRIPTION
**Pull request by miguelgarcia**
* Added mission execution capabilities. 
* Implemented Missions Module and Missions Executor

The executor receives commands from the cloud in the form of messages.

See design doc [Missions Executor](https://docs.google.com/document/d/1QqZkqa1LEl8xFFDZ3ygq_uZsB0fNwrDURylSjxmDWG0/edit#heading=h.ibkxl4cmv68t)

Demo: 

https://user-images.githubusercontent.com/398428/234882942-7f42ff26-c370-4a47-98db-8c1016fc7d8a.mp4

### CODE REVIEW

**Comment by miguelgarcia** [comment]
Suggestion for reviewers: Start by looking at the demo and test before diving into the implementation

**Comment by miguelgarcia** [comment]
@leandropineda I implemented missions support in the SDK. The implementation is a little big, the design doc https://docs.google.com/document/d/1QqZkqa1LEl8xFFDZ3ygq_uZsB0fNwrDURylSjxmDWG0/edit#heading=h.ibkxl4cmv68t explains some details about how it works.

Can you review?

File: inorbit_edge/missions.py [hunk 1]
```diff
@@ -0,0 +1,537 @@
+# This module provides mission execution capabilities
+# See design doc https://docs.google.com/document/d
```
Comment (leandropineda) [inline inorbit_edge/missions.py:2]: I wouldn't include links to design docs on a public repo

File: inorbit_edge/missions.py [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+
+class MissionsModule:
+    """
+    MissionsModule acts as the interface between the robot session and the
+    MissionExecutor. Its main responsibility is handling commands from the robot
+    session, parsing them and passing them to the executor.
+    """
+
+    def __init__(self, robot_session):
+        self.logger = logging.getLogger(__class__.__name__)
```
Comment (leandropineda) [inline inorbit_edge/missions.py:49]:
```suggestion
        self.logger = logging.getLogger(self.__class__.__name__)
```

File: inorbit_edge/missions.py [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+
+class MissionExecutor:
+    """
+    MissionExecutor handles missions execution and enforces execution rules, like:
+     - Can only run a mission if no mission is running
+     - Can only cancel the current mission
+    """
+
+    def __init__(self, robot_session):
+        self.logger = logging.getLogger(__class__.__name__)
```
Comment (leandropineda) [inline inorbit_edge/missions.py:123]:
```suggestion
        self.logger = logging.getLogger(self.__class__.__name__)
```

File: inorbit_edge/missions.py [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+                self.current_step_idx = step_idx
+                self.current_step = self.steps[step_idx]
+                self.report()
+            # HACK(mike) sending two reports without waiting can cause issues
+            # in the backend
+            time.sleep(5)
+            try:
+                self.current_step.execute(self)
+                if not self.current_step.success():
+                    raise Exception()
```
Comment (leandropineda) [inline inorbit_edge/missions.py:227]: Raising bare Exceptions is not recommended. I'd change it for a less general `RuntimeError` exception.

I'd also consider defining our own exceptions to improve readability e.g.

```
class FailedMissionStepException(Exception):
    "Raised when mission step fails to execute"
    pass
```

**Comment by miguelgarcia** [comment]
I addressed the feedback @leandropineda . Can you take another look? Thanks!

**Review by leandropineda — APPROVED** [review]

**Pull request merged**